### PR TITLE
Add Review Status To SGTM Sync

### PR DESCRIPTION
**Pull request by drewalam-asana**
- added review status CF to SGTM sync. There are 4 possible values: Changes Requested, Not Ready, Approved, Needs Review. This will help folks figure out if a PR needs to be reviewed or not.
- modified the approved status for Pull Request objects to utilize the latest review's status to determine if a PR is considered to be "Approved" or "Changes Requested"

### Testing
General format
```
In [1]: import src.github.controller as github_controller; import src.github.graphql.client as graphql_client

In [2]: pull_request = graphql_client.get_pull_request("PR_kwDOAr0gsc5elLoP")

In [3]: github_controller.upsert_pull_request(pull_request)
```

I tested the controller with a [PR ](https://github.com/Asana/codez/pull/224261)I already had up that had the changes requested status. I was able to modify the Asana task to reflect this new status (changes request).
I then asked someone else to approve the pr changes to approved.
I then asked someone else to request changes and the pr status changes and it goes back to needs changes. Task logs:
image

I created a draft PR too and the Draft status was also set. Also tested going from needs review -> draft on the pr and works as expected.

Pull Request synchronized with [Asana task](https://app.asana.com/0/0/1205891141351354)

### CODE REVIEW

File: src/asana/helpers.py
Comment (suzyng83209) [inline src/asana/helpers.py:102]: Does this account for the state where one person requested changes and another person approves later?

SGTM currently waits for all reviewers that requested changes to approve before it determines that the PR has been "approved" instead of looking at the last review. You can checkout `pull_request.is_approved` logic

Comment (drewalam-asana) [reply src/asana/helpers.py:102]: I don't think this makes sense to wait for everyone to approve the PR to mark the PR as approved. 
I think it makes more sense to refer to the latest review left on the PR. What if user A requests changes that are not actually needed and user B follows up saying it's okay and approves the changes. If user B does agree with user A's comments, they'll just add in their additional comments or also request changes as well.

Thoughts?

**Comment by drewalam-asana** [comment]
Current dependencies on/for this PR:
* master
  * **PR #158** <a href="https://app.graphite.dev/github/pr/Asana/SGTM/158?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Asana/SGTM/158?utm_source=stack-comment).

File: scripts/setup_sgtm_tasks_project.py
Comment (suzyng83209) [inline scripts/setup_sgtm_tasks_project.py]: 🤔 we already have a `Draft` enum for `PR Status`. Can this just be unpopulated if its a draft?
image

**Review by suzyng83209 — CHANGES_REQUESTED** [review]

**Review by suzyng83209 — APPROVED** [review]
LGTM! Thanks for doing all the user research and following up with the different avenues I suggested. 

One thing to double check here is error handling. Will the task update/creation fail if the project doesn't have the custom field? If so, we probably need to update all SGTM projects to have this custom field before we deploy this change

**Comment by drewalam-asana** [comment]
When I accidentally tried to test this PR on an Asana tasks for the SGTM repo, it didn't populate correctly so it's safe. I'll go in and modify every SGTM project to have this field though.

**Pull request merged**
